src/msspreadsheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,639 @@
+                                            self.properties.workbookName + ".xlsx:/workbook/worksheets/" + 
+                                            self.properties.worksheetName + "/tables/" + tableName), request);
+
+        if response is error {
+            HttpError httpError = error(HTTP_ERROR, message = "Error occurred while accessing the Microsoft Graph API.", 
+                                        errorCode = HTTP_ERROR, cause = response);
+            return httpError;
+        }
+
+        http:Response httpResponse = <http:Response> response;
+
+        if httpResponse.statusCode != http:STATUS_OK {
+            HttpResponseHandlingError httpResponseHandlingError = error(HTTP_RESPONSE_HANDLING_ERROR,
+                message = "Error occurred while inserting data into table.", errorCode = HTTP_RESPONSE_HANDLING_ERROR);
+            return httpResponseHandlingError;
+        }
+
+        //If the table exists we will get a JSON response with the table's information
+        json|error responseJson = httpResponse.getJsonPayload();
+
+        if !(responseJson is map<json>) {
+            typedesc<any|error> typeOfRespone = typeof responseJson;
+            TypeConversionError typeError = error(TYPE_CONVERSION_ERROR, 
+                    message = "Invalid response; expected a `map<json>` found " +  typeOfRespone.toString(), 
+                    errorCode = TYPE_CONVERSION_ERROR);
+            return typeError;
+        }
+
+        map<json> payload = <map<json>> responseJson;
+    
+        json|error identifier = payload.id;
+        if !(identifier is string) {
+            typedesc<any|error> typeOfIdentifier = typeof identifier;
+            TypeConversionError typeError = error(TYPE_CONVERSION_ERROR, 
+                    message = "Invalid identifier; expected a `string` found " +  typeOfIdentifier.toString(), 
+                    errorCode = TYPE_CONVERSION_ERROR);
+            return typeError;
+        }
+
+        string sheetIdentifier = <string> identifier;
+
+        //Address is not returned from the above API call. Hence the address is initialized to an empty string
+        string address = "";
+
+        Table resultsTable = <@untainted> new (self.microsoftGraphConfig, self.properties.path, 
+                    self.properties.workbookName, self.properties.sheetId, self.properties.worksheetName,
+                    sheetIdentifier, address, tableName);
+
+        return resultsTable;
+    }
+};
+
+# Table Client Object.
+# + tableClient - HTTP client endpoint for the table
+# + properties - table specific properties
+public type Table client object {
+    http:Client tableClient;
+    TableProperties properties;
+
+    public function __init(MicrosoftGraphConfiguration msGraphConfig, string path, string workbookName, string sheetId,
+                            string worksheetName, string tableID, string address, string tableName) {
```

Comment:
Wrong indentation? Practice is to add 4 spaces from the original column, when function parameters are in the next line and no spaces from the original column, if the return statement is in the next line.

Reply:
Fixed all the formatting issues using the VS Code code formatting feature